packages/@react-stately/numberfield/src/useNumberFieldState.ts

```diff
@@ -261,6 +305,44 @@ export function useNumberFieldState(
 
   let validate = (value: string) => numberParser.isValidPartialNumber(value, minValue, maxValue);
 
+  let parseValueInAnySupportedLocale = (value: string) => {
```

Comment:
would it be better to let the paste happen, then when numberfield re-renders and formats the value, we could live announce then?
Something along these lines
```
let shouldAnnouncePaste = useRef(false)
useEffect(() => {
  if (shouldAnnouncePaste.current) {
    // announce logic here
    shouldAnnouncePaste.current = false;
  }
);
```

Reply:
That would address the problem presented in #4967 where the number "3,000,000.25" becomes "3" when pasted into a form in a locale with different grouping or decimal characters.

Reply:
ah, thank you for reminding me of that conversation from last year, I'd forgotten about it
I see what you're trying to do now